database/tables/client_fingerprints.pg

```diff
@@ -0,0 +1,23 @@
+columns
+    accept_language: character varying(255)
```

Comment:
Can we make this `NOT NULL`? Similarly for `user_agent`, `user_id`, and `user_session_id`.

Reply:
I attempted to make this not null and was failing tests due to the tests that do not have an ip and user agent attached to them. Is there a better way to test these and allow them to be `NOT NULL`?

Reply:
Good question. @nwalters512 do you have any idea how to get the tests to act more like a real browser? Alternatively, in the code that creates the fingerprints we could check `config.devMode` and if we are in dev mode then we could fill in an IP and other info if it's missing?

Reply:
I would expect tests in requests to have an IP (should just be localhost, right?). That said, there is no guarantee that requests will have an `Accept-Language` header, or actually _any_ of these headers. I think we have to keep the headers nullable.

Reply:
That's fair. @mylesw27 is the IP non-null at least? I'm happy to keep the other columns nullable.

Reply:
Yes, currently the IP is non-null.